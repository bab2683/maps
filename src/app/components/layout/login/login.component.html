<app-input type="email" placeholder="Email" name="email" label="Your email"></app-input>
<app-input
  class="pass"
  label="Your password"
  name="password"
  placeholder="Password"
  type="password"
></app-input>
