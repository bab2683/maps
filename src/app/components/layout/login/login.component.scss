.pass {
  margin-top: 10px;
}
