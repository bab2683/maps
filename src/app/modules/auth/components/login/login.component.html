<h3>Login</h3>
<bab-form [form]="form"></bab-form>
