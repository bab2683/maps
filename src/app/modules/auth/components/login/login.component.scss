@import "common";

button {
  margin-left: $base-spacing;
}

.form {
  display: inline-block;
}
