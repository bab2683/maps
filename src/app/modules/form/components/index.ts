export * from './composer';
export * from './field';
export * from './input';
