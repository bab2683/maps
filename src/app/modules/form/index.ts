export * from './form.module';
export * from './interfaces';
