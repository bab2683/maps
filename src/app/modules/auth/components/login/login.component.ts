import { Component, EventEmitter, OnInit, Output, ViewChild } from '@angular/core';
import { EmailValidator, Validators } from '@angular/forms';

import { ComposerComponent, FormResult, FormRow } from '@mod/form';
import { AuthLoginMail } from '../../auth.model';

@Component({
  selector: 'auth-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss']
})
export class LoginComponent implements OnInit {
  @ViewChild('form', { static: false }) public form: ComposerComponent;
  rows: FormRow[];

  @Output() loginData: EventEmitter<AuthLoginMail> = new EventEmitter();

  constructor() {
    this.rows = [
      {
        fields: [
          {
            label: 'Your email',
            name: 'email',
            type: 'email',
            validators: [new EmailValidator(), Validators.required]
          },
          {
            label: 'Your password',
            name: 'password',
            type: 'password',
            validators: [Validators.required]
          }
        ]
        // sizes: {
        //   s: 2
        // }
      }
    ];
  }

  public ngOnInit(): void {}

  public emailAndPasswordLogin(): void {
    const result: FormResult | null = this.form.submit();
    if (result) {
      this.loginData.emit({ email: result.email, password: result.password });
    }
  }
}
