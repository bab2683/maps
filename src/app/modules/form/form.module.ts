import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

import { ComposerComponent, FieldComponent, InputComponent } from './components';

@NgModule({
  declarations: [ComposerComponent, InputComponent, FieldComponent],
  exports: [ComposerComponent],
  imports: [CommonModule, FormsModule, ReactiveFormsModule]
})
export class FormModule {}
