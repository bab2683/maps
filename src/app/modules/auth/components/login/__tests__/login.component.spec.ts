import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginComponent } from '../login.component';
import { FormModule } from '@mod/form';

describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;
  let loginDataSpy: any;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LoginComponent],
      imports: [FormModule],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    loginDataSpy = jest.spyOn(component.loginData, 'emit');
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('emailAndPasswordLogin', () => {
    it('should emit the values if form is valid', () => {
      const result = { email: '', password: '' };
      jest.spyOn(component.form, 'submit').mockReturnValue(result);
      component.emailAndPasswordLogin();
      expect(loginDataSpy).toBeCalledWith(result);
    });
  });
});
