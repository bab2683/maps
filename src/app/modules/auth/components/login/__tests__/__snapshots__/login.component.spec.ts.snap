// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match snapshot 1`] = `
<auth-login
  form={[Function ElementRef]}
  loginData={[Function EventEmitter]}
  rows={[Function Array]}
>
  <bab-form /><button
    class="m--l"
    type="button"
  >
    Continue
  </button>
</auth-login>
`;
